uniffi_bindgen/src/bindings/kotlin/templates/CallbackInterfaceRuntime.kt

```diff
@@ -9,24 +9,26 @@ internal const val UNIFFI_CALLBACK_UNEXPECTED_ERROR = 2
 /**
  * @suppress
  */
-public abstract class FfiConverterCallbackInterface<CallbackInterface: Any>: FfiConverter<CallbackInterface, Long> {
+public abstract class FfiConverterCallbackInterface<CallbackInterface: Any>(
+    internal val langIndex: Int
+): FfiConverterRustBuffer<CallbackInterface> {
     internal val handleMap = UniffiHandleMap<CallbackInterface>()
 
     internal fun drop(handle: Long) {
         handleMap.remove(handle)
     }
 
-    override fun lift(value: Long): CallbackInterface {
-        return handleMap.get(value)
+    override fun read(buf: ByteBuffer): CallbackInterface {
+        val handle = buf.getLong()
+        assert(buf.getInt() == this.langIndex) { "Callback interface has been called in the wrong language" }
```

Comment:
This isn't true in all cases: we should be able to call from one language to another via Rust.

Reply:
Yes, I think there's 2 options here:
- Make it so the Rust scaffolding functions work with foreign handles as well.  If Kotlin calls the scaffolding function with a handle from React-native, then Rust will forward the call using the React-native vtable
- Make everything go through vtables.
  - Don't export the Rust scaffolding functions directly
  - Add a function that inputs a language index and returns the vtable for that language.
  - Foreign code uses that to get a vtable whenever they see a handle with a language index they don't recognize.  Once it has a vtable, it uses that to make the call.  This works the exactly same for Rust and other foreign languages.

The second option seems better for performance and I kind of like not special-casing Rust.  I might be a pain to implement though.